### ISSUE TITLE
Source maps not being deleted from the web user interface

### ISSUE THREAD

**Issue by YoofiBP** [getsentry/sentry#28762]
### Environment

SaaS (https://sentry.io/)

### Steps to Reproduce

1 Log into Sentry Admin Dashboard
2 Select Dashboards on the side bar menu
3 Select Processing -> Sourcemaps
4 Attempt to delete a source map entry

### Expected Result

Source map entry should be deleted and not visible in the UI

### Actual Result

All source maps deleted are still present

<img width="1204" alt="Screenshot 2021-09-22 at 3 28 35 PM" src="https://user-images.githubusercontent.com/34424854/134373778-bbf8ece5-cbd3-4bc4-9eec-7cb817db4d3a.png">

**Comment by getsentry-release** [comment]
Routing to @getsentry/owners-ingest for [triage](https://develop.sentry.dev/processing-tickets/#3-triage). ⏲️

**Comment by jjbayer** [comment]
Thanks @YoofiBP for reporting this, we will investigate it.

**Comment by jjbayer** [comment]
@YoofiBP Pressing the "delete" button for an entry on this page does not actually remove the entry, but only the artifacts contained in it (If you hover over the button, you will see it says "Remove All Artifacts"). Can you confirm that the artifact count drops to zero after deleting?

I realize this is not ideal user experience, but there's a performance reason for displaying entries with 0 artifacts (see also https://github.com/getsentry/sentry/pull/19505).

**Comment by iambriccardo** [comment]
Hi @YoofiBP, since we are releasing a new system for sourcemaps we also took our time to address a better UI for showing that no release bundle is connected to that release. Unfortunately, because the same technical issues persist this is the best solution we were able to come up with. In addition, since we are going to fade out the release bundle system once the new one is well-tested, it didn't make much sense to allocate a considerable amount of engineering effort toward solving the problem at its core.
